source/game/field/obj/ObjectCollidable.hh

```diff
@@ -15,6 +15,7 @@ public:
 
 protected:
     virtual void createCollision();
+    virtual EGG::Vector3f collisionCenter() const;
```

Comment:
```suggestion
    virtual const EGG::Vector3f &collisionCenter() const;
```

If we make the body of each `collisionCenter` override have a `static const EGG::Vector3f` that specifies what we want to return, then we can initialize that once, and avoid this copy constructor call every time we need to ask for the collision center. As for `EGG::Vector3f::zero`, that already allows for us returning a reference to it.

Reply:
You're right - we should make this a const reference. However, the overrides don't require static instances, as they can be members of the class if necessary.